src/Features/LanguageServer/Protocol/Handler/SemanticTokens/TokenModifiers.cs

```diff
@@ -16,5 +16,6 @@ internal enum TokenModifiers
         None = 0,
         Static = 1,
         ReassignedVariable = 2,
+        ObsoleteSymbol = 4,
```

Comment:
If you update https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=/src/product/RemoteLanguage/Impl/Features/SemanticTokens/Extensions/RoslynSemanticTokenTypes.cs&version=GBdevelop&line=89&lineEnd=90&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents this should work in VS when LSP is enabled (ie, in Razor files 😁)

Reply:
imho we should use the standard lsp 'deprecated' token modifier, then I don't believe we will have to update that list in the client.  (there may be work on the vs lsp client to actually render something for the deprecated tag though).

Reply:
I completely agree with you about using the standard modifier, but right now I think that list needs to be updated anyway. The platform just falls back to "text" for anything not in that dictionary: https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=/src/product/RemoteLanguage/Impl/Features/SemanticTokens/Tagger/SemanticTokensTaggerBase.cs&version=GBdevelop&line=626&lineEnd=627&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

I don't think VS has a very standards compliant implementation right now.

Reply:
➡️ Rewriting this based on https://github.com/dotnet/roslyn/pull/72156#discussion_r1498165071

Reply:
That's good, using the standard modifier is better, but just to be clear, it won't make this feature work in VS LSP without a change to the file in my first link.